### ISSUE TITLE
A detailed plan for 08/12/20

### ISSUE THREAD

**Issue by T-Wisse** [T-Wisse/MEP_Thomas#7]
In the meeting of 01/12 we said I would have a more detailed plan the coming meeting. I am thinking about what this plan would look like and what it should contain. I think it should resemble a clear and concise summary of my approach and goals for the coming time, say until I start in the lab. 

It should have the exact data I want to use listed and where it comes from, (such as the list of genes associated with GO slim terms from yeastmine). Then it should also explain how I will use that data, which will first be looking for functional modules in which there is a strong correlation of multiple attributes (such as common number of go terms) with the type of genetic interaction by using your code. Once I have one or more modules to work with I will create predictions of the type of genetic interaction between 2 genes based on these attributes and check how well that works.

For the prediction part I am still not sure what would be a good way to do it. I think a classifier trained for a specific module may work but if you have other suggestions they are very welcome. Do you feel like this is what Liedewij has in mind and if anything is missing/should be different?

**Comment by leilaicruz** [comment]
yes I think what is useful are bullet points of specific tasks that obey certain goal you have in mind (not a super polish goal , just something that have a minimum of sense).  Also the more concise the tasks are , the more easy is to test them afterwards. 
When doing your plan keep things simple and concise .

**Comment by leilaicruz** [comment]
Can you publish the plan you discuss with Liedewij here? and we can later in the meeting go through it to see how useful it was to you :)

**Comment by T-Wisse** [comment]
![planDecember](https://user-images.githubusercontent.com/70693768/105042868-0d85c780-5a65-11eb-9272-e2088e9342a4.png)
This was it. I think it was already ambitious but I also overestimated my python ability, so I did not get as far as I hoped

**Comment by leilaicruz** [comment]
Write here a reflection on what you did from the plan and the learning you got from these activities. What you did not achieve , and why. And what would be your next steps in the short term , based on the scope of your project. 
With the discussion of this reflection we will close this issue.

**Comment by T-Wisse** [comment]
## Reflection 

### Achieved
- I created 2 lists of genes which make up functional clusters based on a common GO term. One for cell polarity and one for protein folding. More functional clusters can be made simply by editing the GO term in the script [getGenesFromGOTerm.py](/T-Wisse/MEP_Thomas/blob/main/src/getGenesFromGOterm.py "Hi") and running the script. With that the first step is complete.
- I used attribute data on GO terms and common interactors that was already downloaded.
- I plotted pairplots of attribute data vs interaction type. Both for individual genes and for all genes in the cluster with [script_common_go_for_interactors.py](/T-Wisse/python-modules-for-bioinformatic-analyses/blob/project_code/src/script_common_go_for_interactors.py) .

### Not achieved
- I started on a script to get common pathways of genes and a single script to generalize getting any feature with data from yeastmine but did not finish this. 
- Determine correlation of attributes with interaction type
- Making and evaluating bayesian classifier

### Why not achieved
- I put the common pathway script and general script for getting a feature on hold when I realised how complicated it would be and therefore how much time it would cost me. It makes more sense to have seperate functions to get specific things from yeastmine when I need them.
- Determining any correlations is more difficult than I expected. It varies per gene. Furthermore most individual genes have a low amount of interactors which probably causes a lot of this variation.
- Without a correlation next steps are not relevant
- Persistent issues with resulting data not being completely as expected. What ran fine first later gave differernt results. 

### What have I learned
- While generalizing code is great, it is not always convenient to do so. I have to think about if it makes sense
- Python skills. This will probably be very useful in the future.
- We can not get useful results about features from how individual genes interact due to low interactor count. We should use functional clusters.
- I need to be able to check the results 
- It can be easier to make my own script than to understand and continuously attempt to understand another person's script.
- I should make more/better use of git for version control. I made some changes that later turned out to negatively affect the code, but as I had not consitently commited and pushed I could not go back.
- A discussion with Enzo based on my presentation at the labmeeting alerted me to the hieracrchical nature of GO terms and that this means we can better estimate how common two genes are in function by taking the children of GO terms into account

### Next steps
- Use common 1st layer children of GO terms 
- Find a way to determine correlations between features and interaction type (discuss with Leila/someone else who might have ideas about hwo to do this)
- Use newfound python skills to continue on Wessels work (open new issue on this)

**Comment by leilaicruz** [comment]
> I created 2 lists of genes which make up functional clusters based on a common GO term. One for cell polarity and one for protein folding. More functional clusters can be made simply by editing the GO term in the script <getGenesFromGOTerm.py> and running the script. With that the first step is complete.
> I used attribute data on GO terms and common interactors that was already downloaded.
> I plotted pairplots of attribute data vs interaction type. Both for individual genes and for all genes in the cluster.

- Can you provide links of where is the code that does this? 
- After the plot , what did you learn about the interplay between the chosen attributes and the interaction type between genes?

**Comment by leilaicruz** [comment]
> Find a way to determine correlations between features and interaction type (discuss with Leila/someone else who might have ideas about hwo to do this)

This was an exercise for you to get acquainted with the topic and to also train python skills on a relevant topic, yet I understand is complicated. 

I have scripts that do some correlations with common interactors and interaction type, and with common protein domains and synthetic lethality interaction , that maybe give you ideas , but remember this is all in the context of finding which genes relationships that can be known in advance can be useful to predict their interaction type , for example synthetic lethality. 
- [from protein domains to lethality](https://github.com/leilaicruz/python-modules-for-bioinformatic-analyses/blob/master/src/script_ML_pipeline_protein_domains2lethality.py)
- [common interactors to interaction type](https://github.com/leilaicruz/python-modules-for-bioinformatic-analyses/blob/master/src/script_common_interactors_partners.py)
